onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -1270,31 +1282,20 @@ Applies to session load, initialization, etc. Default is 0.)pbdoc")
           [](const PySessionOptions* options, const char* config_key) -> std::string {
             const std::string key(config_key);
             std::string value;
-            if (!options->config_options.TryGetConfigEntry(key, value))
+            if (!options->value.config_options.TryGetConfigEntry(key, value))
               throw std::runtime_error("SessionOptions does not have configuration with key: " + key);
 
             return value;
           },
           R"pbdoc(Get a single session configuration value using the given configuration key.)pbdoc")
       .def(
           "register_custom_ops_library",
-          [](PySessionOptions* options, const char* library_path) -> void {
+          [](PySessionOptions* options, const char* library_name) -> void {
 #if !defined(ORT_MINIMAL_BUILD) || defined(ORT_MINIMAL_BUILD_CUSTOM_OPS)
-            // We need to pass in an `OrtSessionOptions` instance because the exported method in the shared library expects that
-            // Once we have access to the `OrtCustomOpDomains` within the passed in `OrtSessionOptions` instance, we place it
-            // into the container we are maintaining for that very purpose and the `ortSessionoptions` instance can go out of scope.
-            OrtSessionOptions s;
-
-            options->custom_op_libraries_.emplace_back(std::make_shared<CustomOpLibrary>(library_path, s));
-
-            // reserve enough memory to hold current contents and the new incoming contents
-            options->custom_op_domains_.reserve(options->custom_op_domains_.size() + s.custom_op_domains_.size());
-            for (size_t i = 0; i < s.custom_op_domains_.size(); ++i) {
-              options->custom_op_domains_.emplace_back(s.custom_op_domains_[i]);
-            }
+            OrtPybindThrowIfError(options->RegisterCustomOpsLibrary(ToPathString(library_name)));
```

Comment:
Note that the previous implementation of this method passed a default-constructed `OrtSessionOptions` to custom op libraries. Now, we're passing a pointer to the _actual_ `PySessionOptions` to custom op libraries, which is more consistent with a user's expectations (imo).

This change of particular importance to https://github.com/microsoft/onnxruntime/pull/13427 because it allows custom op libraries to extract session configuration entries from the passed in session options. Without this change, a custom op library would only ever see an "empty" session options object.